File: golang/q02/part2/part2.go
package part2

import (
	c "aoc2023/q02/common"
	"fmt"
)

func Part2(games []c.Game) {
	sum := 0

	for _, game := range games {
		power := 1
		maxColors := game.MaxColors()

		power = power * maxColors.Red
		power = power * maxColors.Green
		power = power * maxColors.Blue

		sum += power
	}

	fmt.Println("Part 2:", sum)
}

File: js/q02/part2/part2.js
import { parseFile, getGameMaxes } from "../common/index.js";

function part2(path) {
  const parseFileResult = parseFile(path);
  if (parseFileResult.isErr()) {
    console.error(`Parse file error: ${parseFileResult.unwrap()}`);
    return;
  }

  const result = parseFileResult.unwrap().reduce((accumulator, game) => {
    const maxes = getGameMaxes(game);
    return accumulator + (maxes.red * maxes.green * maxes.blue);
  }, 0);

  console.log(`Part 2: ${result}`);
}

part2('./data/input.txt');
